docs/zello-sdk-documentation/com/zello/sdk/Events.html

```diff
@@ -282,6 +289,32 @@ <h2 title="Interface Events" class="title">Interface Events</h2>
 </dl>
 </li>
 </ul>
+<a name="onMicrophonePermissionNotGranted--">
+<!--   -->
+</a>
+<ul class="blockList">
+<li class="blockList">
+<h4>onMicrophonePermissionNotGranted</h4>
+<pre>void&nbsp;onMicrophonePermissionNotGranted()</pre>
+<div class="block"><p>
+     Called when an invocation of the <a href="../../../com/zello/sdk/Zello.html#beginMessage--"><code>Zello.beginMessage()</code></a> method failed because the microphone permission hasn't been granted.
+ </p>
+ <p>
+     This method is invoked on the UI thread.
+ </p>
+ <p>
+     The normal use case for handling this error would be to call <a href="../../../com/zello/sdk/Zello.html#showMicrophonePermissionDialog--"><code>Zello.showMicrophonePermissionDialog()</code></a>.
+     However, it is the responsibility of the app using the SDK to determine if they should handle this error or not.
+     For example, if there are multiple apps using the SDK, they will all receive this this callback. The <a href="../../../com/zello/sdk/Zello.html#showMicrophonePermissionDialog--"><code>Zello.showMicrophonePermissionDialog()</code></a>
+     method should likely only be on one of these apps (the one in the foreground).
```

Comment:
Thought: The client app today is instructed to call enterPowerSavingMode() when going into the background. The Sdk could check if it is currently in power saving mode when receiving this showMicrophonePermissionDialog() call and ignore it if it is. This would remove the burden from the client from tracking whether or not it should make the call.